### PR TITLE
rename AssetCheckHandle -> AssetCheckKey

### PR DESCRIPTION
**Pull request by johannkm**
Context https://github.com/dagster-io/dagster/pull/16672#discussion_r1334574671

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16790** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16790" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16790?utm_source=stack-comment).

File: python_modules/dagster-graphql/dagster_graphql/schema/asset_checks.py [hunk 1]
```diff
@@ -205,5 +205,5 @@ class GrapheneAssetCheckHandle(graphene.ObjectType):
     class Meta:
         name = "AssetCheckhandle"
```
Comment (johannkm) [inline python_modules/dagster-graphql/dagster_graphql/schema/asset_checks.py:206]: will rename once we don't have js diffs in flight

File: python_modules/dagster/dagster/_core/definitions/asset_check_spec.py [hunk 2]
```diff
@@ -28,8 +28,8 @@ class AssetCheckSeverity(Enum):
 
 
 @experimental
-@whitelist_for_serdes
-class AssetCheckHandle(NamedTuple):
+@whitelist_for_serdes(old_storage_names={"AssetCheckHandle"})
```
Comment (johannkm) [inline python_modules/dagster/dagster/_core/definitions/asset_check_spec.py:31]: I believe this handles all backcompat concerns

File: python_modules/dagster-graphql/dagster_graphql/schema/asset_checks.py (hunk 1 above)
Comment (johannkm) [reply python_modules/dagster-graphql/dagster_graphql/schema/asset_checks.py:206]: nice typo too

**Review by sryza — APPROVED** [review]

**Pull request merged**